### DISCUSSION TITLE
Export kiali graph to yaml file

### DISCUSSION THREAD

**Discussion by phamquiluan** [kiali/kiali#6211] [General]
I am wondering how can I generate a yaml file that stores the content of the graph (nodes, edges, connection, and other information).

It will be very good for other algorithms to perform root cause analysis or performance debugging.

![image](https://github.com/kiali/kiali/assets/24642166/65d5af42-8ea3-4930-8036-57bf290e4890)

**Comment by aljesusg** [comment]
Moved to discussion, this is not an issue.
Can you tell us the use case?

**Reply by phamquiluan** [reply]
Thank you for your support @aljesusg, here is one use case.

Problem: Root cause analysis. 
Context: Given a microservice system with N services. A service X has a fault at time t (CPU hog, memory leak, etc) that leads to an SLO violation in the front-end service. The task is to find X
Input: metrics, the graph

Solution:
1. calculate the anomalous score for each metric
2. correlate them to find the root cause
3. having the graph will reduce the computational efforts

I hope you can find it meaningful. Thank you so much for building this project.

**Comment by jmazzitelli** [comment]
1. From the Kiali UI, click the help icon at the top in the masthead (the circle with the question mark in it) to get a dropdown menu:

![image](https://github.com/kiali/kiali/assets/2029470/775b6351-dde1-4b43-8190-26acb5f425a1)

2. Click "View Debug Info" found in that dropdown menu.
3. In the popup window that results, select the "Additional State" tab.

At this point, you will see JSON where the graph (edges and nodes) can be found. The graph data starts at the JSON element "reduxState.graph".

4. Click the "Download" button to download the JSON which you can then process yourself using whatever tools you want.

![image](https://github.com/kiali/kiali/assets/2029470/96c5921b-dd28-48b5-8592-db6f92e3f4d6)

**Reply by phamquiluan** [reply]
It works! Thank you so much 

```
(cmd)luan@kitty:~/ws/kiali$ python3 main.py 
cartservice -> redis-cart
checkoutservice -> cartservice
checkoutservice -> shippingservice
checkoutservice -> emailservice
checkoutservice -> paymentservice
checkoutservice -> currencyservice
checkoutservice -> productcatalogservice
recommendationservice -> productcatalogservice
frontend -> adservice
frontend -> cartservice
frontend -> checkoutservice
frontend -> recommendationservice
frontend -> shippingservice
frontend -> currencyservice
frontend -> productcatalogservice
loadgenerator -> frontend
```
